File: Money.API/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace Money.API.Models
{
    public class Transaction
    {
        public int Id { get; set; }

        [Required]
        [StringLength(200)]
        public string? Description { get; set; }

        [Required]
        public DateOnly Date { get; set; }

        public int CategoryId { get; set; }

        [Required]
        public virtual Category Category { get; set; }

        [Required]
        public decimal Debit { get; set; }

        [Required]
        public decimal Credit { get; set; }
    }
}

File: Money/src/app/models/transaction.ts
import { Category } from "./category";

/**
 * Transaction from National Bank of Canada CSV file.
 */
export interface Transaction {
    /**
     * unique identifier.
     */
    id: number;

    /**
     * Description of the transaction.
     */
    description: string;

    /**
     * Date of the transaction (without hours).
     */
    date: Date;

    /**
     * unique identifier of the NBC Category
     */
    categoryId: number;

    /**
     * Category.
     */
    category: Category;

    /**
     * Amount debit.
     * 0 if credit > 0.
     */
    debit: number;

    /**
     * Amount credit
     * 0 if debit > 0.
     */
    credit: number
}